src/util/descriptionParser.tsx

```diff
@@ -0,0 +1,49 @@
+import { createElement, ReactNode } from "react";
+
+const RAW_DESCRIPTION_TAG_NAME_MAP: { [k: string]: string } = { b: "strong", i: "em" };
+
+export function rawDescriptionToReact(rawDescription: string): ReactNode[] {
+  return rawDescriptionToReactRecursive(rawDescription).nodes;
+}
+
+function rawDescriptionToReactRecursive(rawDescription: string): {
+  nodes: ReactNode[];
+  stringLeft: string;
+} {
+  const nodes: ReactNode[] = [];
+  let stringLeftToProcess: string = rawDescription;
+  let processedString: string = "";
+
+  for (let i = processedString.length; i < stringLeftToProcess.length; i++) {
+    processedString += stringLeftToProcess[i];
+    const stringChunk = processedString.slice(processedString.length - 3);
```

Comment:
the fact that this slice goes negative fucks with my head.

how extensible will this be to handle the `{dice ...}` case? it looks pretty hard coded to chunks of 3

i think there's a neater way to do this but it needs some thought